frontend/src/charts/TableChart.tsx

```diff
@@ -1,70 +1,135 @@
 import React from "react";
-import { Paper } from "@material-ui/core";
-import Table from "@material-ui/core/Table";
+import {Column, HeaderGroup, Row, usePagination, useSortBy, useTable} from "react-table";
 import TableBody from "@material-ui/core/TableBody";
 import TableCell from "@material-ui/core/TableCell";
-import TableContainer from "@material-ui/core/TableContainer";
 import TableHead from "@material-ui/core/TableHead";
 import TableRow from "@material-ui/core/TableRow";
-import Tooltip from "@material-ui/core/Tooltip";
-import { Row } from "../data/utils/DatasetTypes";
+import TableFooter from "@material-ui/core/TableFooter";
+import TablePagination from "@material-ui/core/TablePagination";
+import TableSortLabel from "@material-ui/core/TableSortLabel";
+import Paper from "@material-ui/core/Paper";
+import {MetricConfig, MetricId} from '../data/config/MetricConfig';
+import {BREAKDOWN_VAR_DISPLAY_NAMES, BreakdownVar} from '../data/query/Breakdowns';
+import {Tooltip} from '@material-ui/core';
 import WarningRoundedIcon from "@material-ui/icons/WarningRounded";
-import {
-  BreakdownVar,
-  BREAKDOWN_VAR_DISPLAY_NAMES,
-} from "../data/query/Breakdowns";
-import { MetricConfig, formatFieldValue } from "../data/config/MetricConfig";
+import TableContainer from '@material-ui/core/TableContainer';
+import Table from '@material-ui/core/Table';
 
 export interface TableChartProps {
-  data: Row[];
+  data: Readonly<Record<string, any>>[];
   breakdownVar: BreakdownVar;
   metrics: MetricConfig[];
 }
 
 export function TableChart(props: TableChartProps) {
-  return (
-    <>
-      {props.data.length <= 0 || props.metrics.length <= 0 ? (
-        <h1>No Data provided</h1>
-      ) : (
-        <TableContainer component={Paper} style={{ maxHeight: "500px" }}>
-          <Table stickyHeader>
-            <TableHead>
-              <TableRow>
-                <TableCell>
-                  {BREAKDOWN_VAR_DISPLAY_NAMES[props.breakdownVar]}
-                </TableCell>
-                {props.metrics.map((metricConfig, i) => (
-                  <TableCell key={i}>
-                    {metricConfig.fullCardTitleName}
+  const {data, metrics, breakdownVar} = props;
+  let columns = metrics.map((metricConfig) => {
+    return {
+      Header: metricConfig.fullCardTitleName,
+      accessor: metricConfig.metricId,
+    };
+  });
+  columns = [{
+    Header: BREAKDOWN_VAR_DISPLAY_NAMES[breakdownVar],
+    accessor: breakdownVar as MetricId,
+  }].concat(columns);
+
+  // eslint-disable-next-line
```

Comment:
I may add a note what this is escaping. If someone else modifies the line later it would be easier for them to know if the lint escape is still necessary